### PR TITLE
74919 - ess api response when aio feature on/off

### PR DESCRIPTION
**Pull request by MastekRohitChettri**
- Done code changes
- Added unit test cases
- Functional test cases are added

Note: This branch consists PBI 74325 - ESS API Response Structure when AIO feature is ON and AIO cell is requested
         as well.

### CODE REVIEW

**Review by shailen14599 — CHANGES_REQUESTED** [review]

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/appsettings.json [hunk 1]
```diff
@@ -94,7 +94,7 @@
     "CacheProductCode": "AVCS"
   },
   "AioConfiguration": {
-    "AioEnabled": false,
+    "IsAioEnabled": false,
```
[the remark this answers is not in the document]
Comment (MastekRohitChettri) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/appsettings.json]: Resolved

**Review by ray-sj — APPROVED** [review]
looks fine

**Review by shailen14599 — APPROVED** [review]
LGTM

**Review by DanOwenUKHO — CHANGES_REQUESTED** [review]
Looks good, a couple of minor comments.
Stryker report: 44.48% -> 45.38 %

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Models/Response/ExchangeSetResponse.cs [hunk 2]
```diff
@@ -21,8 +21,17 @@ public class ExchangeSetResponse
         [JsonProperty("requestedProductsAlreadyUpToDateCount")]
         public int RequestedProductsAlreadyUpToDateCount { get; set; }
 
+        [JsonProperty("requestedAioProductCount", NullValueHandling = NullValueHandling.Ignore)]
+        public int? RequestedAioProductCount { get; set; } = null;
```
[the remark this answers is not in the document]
Comment (MastekRohitChettri) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Models/Response/ExchangeSetResponse.cs:25]: Based on Aio toggle when toggle is Off then Aio details in response need to hide. So with above code we are removing null properties and when toggle is On then we are assigning above property which returns as Aio details in ess response.

**Pull request merged**